Foundation/FileManager.swift

```diff
@@ -209,20 +240,93 @@ open class FileManager : NSObject {
                 useDarwinPaths = false
             #endif
         }
-        
-        var urls: [URL] = []
-        
+
         for domain in domains {
             if useDarwinPaths {
                 urls.append(contentsOf: darwinURLs(for: directory, in: domain))
             } else {
                 urls.append(contentsOf: xdgURLs(for: directory, in: domain))
             }
         }
-        
+#endif
+
         return urls
     }
-    #endif
+
+#if os(Windows)
+    private class func getURL(for id: KNOWNFOLDERID) -> URL {
+      var pszPath: PWSTR?
+      _ = withUnsafePointer(to: id) { id in
+        SHGetKnownFolderPath(id, DWORD(KF_FLAG_DEFAULT.rawValue), nil, &pszPath)
+      }
+      let url: URL = URL(fileURLWithPath: String(decodingCString: pszPath!, as: UTF16.self), isDirectory: true)
```

Comment:
Does SHGetKnownFolderPath always return nonnil?

Reply:
It returns a HRESULT, we are only interested in the value of `pszPath` which is an out variable.

Reply:
I mean: is the out variable _always_ set? You're not checking for success or failure here.

Reply:
Right, the outer variable is the `HRESULT`.  True, I'm not checking for the error case, since the two known failure cases are: ENOMEM, in which case, death to all!  or the parameter validation failed.  I suppose I can check for the latter if it makes it seem better.

Reply:
I'll add an assertion (`precondition`)